### PR TITLE
PLUMED version 2.8.0 added for GROMACS-2021.5

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)
Requires:
- [x] PR #15640

### CODE REVIEW

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb [hunk 1]
```diff
@@ -9,13 +9,15 @@
...
 # * Dugan Witherick <dugan.witherick@warwick.ac.uk>
 # * Christoph Siegert <christoph.siegert@uni-leipzig.de>
+# * J. Sassmannshausen <j.sassmannshausen@imperial.ac.uk>
 # License::   MIT/GPL
 
 name = 'GROMACS'
 version = '2021.5'
+_plumedver = '2.8.0'
+versionsuffix = '-PLUMED-%s' % _plumedver
 versionsuffix = '-CUDA-%(cudaver)s'
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb:21]:
```suggestion
_cuda_suffix = '-CUDA-%(cudaver)s'
versionsuffix = _cuda_suffix + '-PLUMED-%s' % _plumedver
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]
You are changing the file `GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb` in this PR, but I think you want to create a new file called `GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb`. Is that correct?

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb (hunk 1 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb:21]: Thanks. That slipped my attention. That might explain the wrong name of the EC file. I changed that.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    ('Doxygen', '1.9.1'),
+]
+
+dependencies = [
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+    ('networkx', '2.6.3'),
+    ('PLUMED', _plumedver),
+    ('CUDA', '11.4.1', '', True),
+    ('UCX-CUDA', '1.11.2', versionsuffix),
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb:76]:
```suggestion
    ('UCX-CUDA', '1.11.2', _cuda_suffix),
```

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb (hunk 1 above)
Comment (SebastianAchilles) [reply easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb:21]: Can you also remove the changes to `GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb` from this PR?

Comment (sassy-crick) [reply easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb]: Done (hopefully). I was a bit confused as I had locally the 'correct' filename, i.e. the one I wanted to have. I hope that is now all settled.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb (hunk 2 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb]: Done. We should have thought of it when I done the changes above. Slipped my mind as well.

**Comment by sassy-crick** [comment]
I hope it is all correct now. There should be three EasyConfig files:
for version 2021.3:
`GROMACS-2021.3-foss-2021a-PLUMED-2.7.2.eb` 
for version 2021.5:
`GROMACS-2021.5-foss-2021b-PLUMED-2.8.0.eb` 
`GROMACS-2021.5-foss-2021b-PLUMED-2.8.0-CUDA-11.4.1.eb`

Sorry for the mess, I hope it is all clear now.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-PLUMED-2.8.0.eb [hunk 3]
```diff
@@ -9,25 +9,27 @@
...
 # * Dugan Witherick <dugan.witherick@warwick.ac.uk>
 # * Christoph Siegert <christoph.siegert@uni-leipzig.de>
+# * J. Sassmannshausen <j.sassmannshausen@imperial.ac.uk>
 # License::   MIT/GPL
 
 name = 'GROMACS'
 version = '2021.5'
-versionsuffix = '-CUDA-%(cudaver)s'
+_plumedver = '2.8.0'
+versionsuffix = '-PLUMED-%s' % _plumedver
```
Comment (boegel) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-PLUMED-2.8.0.eb:20]: @sassy-crick You're still changing an existing easyconfig here, which we shouldn't do.

Can you add back `GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb` from `develop` in your PR branch?

**Review by boegel — DISMISSED** [review]

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-PLUMED-2.8.0.eb (hunk 3 above)
Comment (SebastianAchilles) [reply easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-PLUMED-2.8.0.eb:20]: @sassy-crick I have added `GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb` in https://github.com/sassy-crick/easybuild-easyconfigs/pull/8

Comment (sassy-crick) [reply easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-PLUMED-2.8.0.eb:20]: Sorry for the delay in getting back here. Is that now all sorted or are there still problems?

Comment (SebastianAchilles) [reply easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-PLUMED-2.8.0.eb:20]: Thanks a lot! The files look good now :+1: 

I will asked the bot for the test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    'ftp://ftp.gromacs.org/pub/gromacs/',
+]
+sources = [SOURCELOWER_TAR_GZ]
+patches = [
+    'GROMACS-2019_fix_omp_num_threads_and_google_test_death_style_in_tests.patch',
+    'GROMACS-2019_increase_test_timeout_for_GPU.patch',
+    'GROMACS-2021_fix_gmxapi_gmx_allowed_cmd_name.patch',
+    'GROMACS-2021.5_fix_threads_gpu_Gmxapitests.patch',
+    'GROMACS-2021.3_skip_test_for_plumed.patch',
+]
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb:49]:
```suggestion
    'GROMACS-2021_add-missing-sync.patch',
]
```

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    '406f5edd204be812f095a6f07ebc2673c5f6ddf1b1c1428fd336a80b9c629275',
+    # GROMACS-2019_increase_test_timeout_for_GPU.patch
+    '0d16f53d428155197a0ed0b0974ce03422f199d7c463c4a9156a3b99e3c86234',
+    # GROMACS-2021_fix_gmxapi_gmx_allowed_cmd_name.patch
+    'b7ffb292ec362e033db1bedd340353f0644dbaae872127750f3dda1ac7e87d49',
+    # GROMACS-2021.5_fix_threads_gpu_Gmxapitests.patch
+    '7d39da0b431fbc9e94e857552246eb0ec892f26b13e519b3706785cc5e01a563',
+    # GROMACS-2021.3_skip_test_for_plumed.patch    
+    'c06d0a8c3e75b232224f46bc3b70e135c4c02ec4151712970f5edd85d890f018',
+]
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2021.5-foss-2021b-CUDA-11.4.1-PLUMED-2.8.0.eb:62]:
```suggestion
    '52ee257309ff7761c2dd5b26de7dbc63f8ba698082adb88e2843f90e3f9168bf',  # GROMACS-2021_add-missing-sync.patch
]
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]
@sassy-crick Let's include the patch for GROAMCS 2021.x from https://github.com/easybuilders/easybuild-easyconfigs/pull/16026 to fix missing sync in LINCS and SETTLE CUDA kernels

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Comment by sassy-crick** [comment]
Thanks for adding the patches and do that straight away, much appreciated!

**Review by SebastianAchilles — APPROVED** [review]
The PR looks good to me.

**Comment by SebastianAchilles** [comment]
Going in, thanks @sassy-crick!

**Pull request merged**